### PR TITLE
Form: add captionTemplate option to customize group caption

### PR DESCRIPTION
**Pull request by Zedwag**

### CODE REVIEW

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.markup.tests.js [hunk 1]
```diff
@@ -1367,14 +1368,51 @@ QUnit.module('Grouping', () => {
         assert.strictEqual($groups.eq(0).find(`.${labelClass}`).text(), 'First Name: ?', 'Labels\'s content has correct data');
     });
 
+    test('Group caption template should be correctly set as string', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.markup.tests.js]:
```suggestion
    test('Group caption template should be rendered correctly if string is passed', function(assert) {
```

Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.markup.tests.js]: Let's add a separate module?

File: packages/devextreme/scss/widgets/base/_form.scss [hunk 2]
```diff
@@ -123,6 +123,12 @@
   font-size: 20px;
 }
 
+.dx-form-group-custom-caption {
+  .dx-icon {
+    vertical-align: top;
```
Comment (EugeniyKiyashko) [inline packages/devextreme/scss/widgets/base/_form.scss:128]: Why 'top' value is used?

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.markup.tests.js [hunk 3]
```diff
@@ -1367,14 +1368,51 @@ QUnit.module('Grouping', () => {
...
+                {
+                    itemType: 'group',
+                    captionTemplate,
+                }]
+        });
+        const $group = $form.find(`.${FORM_GROUP_CLASS}`);
+        const $caption = $form.find(`.${FORM_GROUP_CUSTOM_CAPTION_CLASS}`);
+
+        assert.ok($caption.parent().is($group), 'caption is placed in group item');
+        assert.strictEqual($caption.text(), captionTemplate, 'caption text is correct');
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.markup.tests.js]: I didn't find the test that check changing captionTemplate option at runtime

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.markup.tests.js [hunk 4]
```diff
@@ -1367,14 +1368,51 @@ QUnit.module('Grouping', () => {
...
+    });
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.form/form.markup.tests.js]: I didn't find the test that check container argument as follows
```
captionTemplate: function(data, container) {
                        assert.deepEqual(isRenderer(container), !!config().useJQuery, 'container is correct');
```

File: packages/devextreme/scss/widgets/base/_form.scss (hunk 2 above)
Comment (Zedwag) [reply packages/devextreme/scss/widgets/base/_form.scss:128]: it's the same value as for dx-icon in custom label. without it icon position looks bad

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by iBat — APPROVED** [review]

**Pull request merged**